lib/connextclient/ConnextClient.ts

```diff
@@ -1045,13 +1084,33 @@ class ConnextClient extends SwapClient {
       units = argUnits.toString();
     }
 
-    const res = await this.sendRequest('/onchain-transfer', 'POST', {
-      assetId: this.getTokenAddress(currency),
-      amount: units,
-      recipient: destination,
+    if (!this.seed) {
+      throw new Error('cannot broadcast transaction without seed');
+    }
+
+    // This is a bridge between the imperative and functional code. First step is to get rid of all of the external state.
+    const broadcastTransaction$ = onChainSendERC20(
+      this.host,
+      this.port,
+      CHAIN_IDENTIFIERS[this.network],
+      this.seed,
+      this.getTokenAddress(currency),
+      destination
+    );
+    broadcastTransaction$.subscribe({
+      // portal pack to imperative world
+      next: (transaction) => {
+        this.logger.info(`on-chain transfer broadcasted, transaction hash: ${transaction.hash}`);
+      },
+      error: (e) => {
+        this.logger.error(`could not broadcast on-chain transfer: ${JSON.stringify(e)}`);
```

Comment:
Should we really use the broadcast terminology? Why not just `on-chain transfer sent, transaction hash: ` and `could not send onchain transfer`?

Reply:
I was thinking we might want to wait for the transfer event for emitting `complete` and `next` when broadcasted. But since complete is not implemented, yet - I'll change this to sent terminology.

Reply:
Changed.

Reply:
Just fyi: those events are emitted when the transaction is included in a block and not when it is accepted in the mempool

Reply:
Yup - good to know. We don't really care about that right now. The swapclient interface will expect `Promise<transactionId: string>`.